app/controllers/users_controller.rb

```diff
@@ -1,3 +1,25 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find params[:id]
```

Comment:
Phân biệt giúp chị `find`, `find_by` và `find_by!` nào

Reply:
- Lệnh "find_by" sẽ tìm và in ra mảng của object đầu tiên theo cặp "key và value" được khai báo và sẽ trả về "nil" nếu những điều kiện đó không xuất hiện.
- Lệnh "find_by!" sẽ tìm và in ra mảng của object đầu theo cặp "key và value" được khai báo và sẽ trả về exception RecordNotFound nếu những điều kiện đó không xuất hiện.
- Lệnh "find" sẽ tìm và in ra mảng của object theo "id" được khai báo và sẽ trả về exception RecordNotFound nếu "id" đó không xuất hiện.
Ngoài ra lệnh "find" có thể thay "id" bằng :first, :last, :all tùy mục đích sử dụng